chain-signatures/node/src/mesh/connection.rs

```diff
@@ -161,14 +162,61 @@ impl Pool {
         self.potential_connections.read().await.clone()
     }
 
-    pub async fn is_participant_stable(&self, participant: &Participant) -> bool {
+    async fn max_block_height_among_participants(&self) -> BlockHeight {
+        self.status
+            .read()
+            .await
+            .values()
+            .filter_map(|state| {
+                if let StateView::Running {
+                    latest_block_height,
+                    ..
+                } = state
+                {
+                    Some(*latest_block_height)
+                } else {
+                    None
+                }
+            })
+            .max()
+            .unwrap_or(0)
```

Comment:
this should just return an Option or Result instead of unwrapping into zero since we end up checking for zero later